Aquib/1_Valid_Sudoku.c.txt

```diff
@@ -0,0 +1,83 @@
+bool checkRow(char **board, int n)
+{
+    char boardstatus[10] = {0};
+    for(int i =0; i < 9;i++)
+    {
+         char cell = board[n][i];
+        if(board[n][i] != '.')
+        {
+            char idx = cell - '0';
+            if(boardstatus[idx] == 0)
+            {
+                boardstatus[idx] = cell;
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+bool checkCol(char **board, int n)
+{
+    char boardstatus[10] = {0};
+    for(int i =0; i < 9;i++)
+    {
+         char cell = board[i][n];
+        if(board[i][n] != '.')
+        {
+            char idx = cell - '0';
+            if(boardstatus[idx] == 0)
+            {
+                boardstatus[idx] = cell;
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+    
+}
+
+bool checkGrid(char **board, int n)
+{
+    char boardstatus[10] = {0};
+    int row = (n /3) * 3;
+    int col = (n % 3) * 3;
+    for(int r = row; r < (row+3);r++)
+    {
+        for(int c = col; c < (col+3);c++)
+        {
+            char cell = board[r][c];
+            if(board[r][c] != '.')
+            {
+              char idx = cell - '0';
+              if(boardstatus[idx] == 0)
+              {
+                boardstatus[idx] = cell;
+              }
+              else
+              {
+                return false;
+              }
+        }
+        }
+    }
+    return true;    
+}
+
+
+
+bool isValidSudoku(char** board, int boardSize, int* boardColSize)
+{
+   for(int n = 0; n < boardSize; n++){
+        if( checkRow(board, n) == false ) return false;
```

Comment:
How about -

if(!checkRow(board, n) || !checkCol(board, n) || !checkGrid(board, n))
    return false;
return true;

Usually '!' is preferred over '== false'

Reply:
Yes. I will change it. I took that approach to provide a bit more clarity. Sorry about that

Reply:
I feel that approach could be helpful for new grads, not for us.